### PR TITLE
8 as a new user i can create a local account

### LINKED ISSUE #8: As a new user, I can create a local account
...so that I can personalise and secure my account

### PR DESCRIPTION
**Pull request by jianyangg**
Close #8; adds the `register` command.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/user/UserRegisterCommand.java
Comment (B-enguin) [inline src/main/java/seedu/address/logic/commands/user/UserRegisterCommand.java:1]: Should this file have unit test done

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 1]
```diff
@@ -96,6 +98,9 @@ public Command parseCommand(String userInput) throws ParseException {
         case UserLogoutCommand.COMMAND_WORD:
             return new UserLogoutCommand();
 
+        case UserRegisterCommand.COMMAND_WORD:
+            return new UserRegisterCommandParser().parse(arguments);
```
Comment (B-enguin) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:116]: Perhaps add testing to the AdressBookParserTest for this new command word

File: src/main/java/seedu/address/model/ModelManager.java [hunk 2]
```diff
@@ -169,6 +173,15 @@ public boolean getUserLoginStatus() {
         return isLoggedIn;
     }
 
+    /**
+     * Returns true if the stored username and password matches the given {@code user}.
+     */
+    @Override
+    public boolean userMatches(User user) {
+        requireNonNull(user);
+        return userPrefs.userMatches(user);
```
Comment (B-enguin) [inline src/main/java/seedu/address/model/ModelManager.java]: Add testing to `ModelManagerTest`

File: src/main/java/seedu/address/model/UserPrefs.java
Comment (B-enguin) [inline src/main/java/seedu/address/model/UserPrefs.java:1]: Add testing for this file

File: src/main/java/seedu/address/model/user/Password.java
Comment (B-enguin) [inline src/main/java/seedu/address/model/user/Password.java:1]: Add testing

File: src/main/java/seedu/address/model/user/User.java
Comment (B-enguin) [inline src/main/java/seedu/address/model/user/User.java:1]: Add testing

**Review by B-enguin — COMMENTED** [review]
Some general feedback

**Review by juliusgambe — DISMISSED** [review]
I think we moved instance creation to the command itself cc: @B-enguin

File: src/main/java/seedu/address/logic/parser/user/UserLoginCommandParser.java [hunk 3]
```diff
@@ -40,7 +40,7 @@ public UserLoginCommand parse(String args) throws ParseException {
         Username username = ParserUtil.parseUsername(argMultimap.getValue(PREFIX_USER).get());
         Password password = ParserUtil.parsePassword(argMultimap.getValue(PREFIX_PASSWORD).get());
 
-        User user = new User(username, password);
+        User user = new User(username, password, true);
```
Comment (juliusgambe) [inline src/main/java/seedu/address/logic/parser/user/UserLoginCommandParser.java:43]: I think instance creation has been relegated to execute method.

File: src/main/java/seedu/address/model/UserPrefs.java [hunk 4]
```diff
@@ -67,6 +82,91 @@ public void setDeliveryBookFilePath(Path deliveryBookFilePath) {
...
+    public static String readFileAsString(String filePath) throws Exception {
+        return new String(Files.readAllBytes(Paths.get(filePath)));
+    }
+
+    /**
+     * Returns the stored user.
+     *
+     * @return storedUser
+     */
+    public User getStoredUser() {
```
Comment (juliusgambe) [inline src/main/java/seedu/address/model/UserPrefs.java:120]: We may need to migrate this JSON Serializable

**Review by juliusgambe — COMMENTED** [review]
General feedback

**Pull request merged**